src/pybind/rados/rados.pyx

```diff
@@ -889,6 +889,13 @@ Rados object in state %s." % self.state)
         if ret >= 0:
             return int(ret)
         elif ret == -errno.ENOENT:
+            rev = self.wait_for_latest_osdmap() 
```

Comment:
no need for a parameter imo - we can always check for the latest map if we get ENOENT. The internal user RadosClient::create_ioctx() already does this manually, so the retry can be removed from that function at the same time.